framework-plugins/lisk-framework-forger-plugin/src/forger_plugin.ts

```diff
@@ -122,11 +157,159 @@ export class ForgerPlugin extends BasePlugin {
+		).toString('base64');
+		const forgerAddressBuffer = Buffer.from(forgerAddress, 'base64');
+
+		return {
+			forgerAddress,
+			forgerAddressBuffer,
+			reward,
+			payload,
+		};
+	}
+
+	private async _incrementForgerInfo(block: string): Promise<void> {
+		const { forgerAddress, reward, payload } = this._getGeneratorAndPayloadInfo(block);
+		const forgerInfo = await this._getForgerInfo(forgerAddress);
+		let isUpdated = false;
+
+		if (this._forgersList.find(forger => forger.address === forgerAddress)) {
+			forgerInfo.totalProducedBlocks += 1;
+			forgerInfo.totalReceivedRewards += BigInt(reward);
+			forgerInfo.totalReceivedFees += await this._getFee(payload, block);
+			isUpdated = true;
+		}
+
+		for (const trx of payload) {
+			if (trx.type === VoteTransaction.TYPE) {
+				for (const vote of (trx.asset as Asset).votes) {
+					const registeredDelegateIndex = this._forgersList.findIndex(
+						forger => forger.address === vote.delegateAddress,
+					);
+					if (registeredDelegateIndex !== -1) {
+						const delegateVoteIndex = forgerInfo.votesReceived.findIndex(aVote =>
+							aVote.address.equals(Buffer.from(vote.delegateAddress, 'base64')),
+						);
+						if (delegateVoteIndex === -1) {
+							forgerInfo.votesReceived.push({
+								address: Buffer.from(vote.delegateAddress, 'base64'),
+								amount: BigInt(vote.amount),
+							});
+						} else {
+							forgerInfo.votesReceived[delegateVoteIndex].amount += BigInt(vote.amount);
+						}
+						isUpdated = true;
+					}
+				}
+			}
+		}
+
+		if (isUpdated) {
+			await this._setForgerInfo(forgerAddress, { ...forgerInfo });
+		}
+	}
+
+	private async _decrementForgerInfo(block: string): Promise<void> {
+		const { forgerAddress, reward, payload } = this._getGeneratorAndPayloadInfo(block);
+		const forgerInfo = await this._getForgerInfo(forgerAddress);
+		let isUpdated = false;
+
+		if (this._forgersList.find(forger => forger.address === forgerAddress)) {
+			forgerInfo.totalProducedBlocks -= 1;
+			forgerInfo.totalReceivedRewards -= BigInt(reward);
+			forgerInfo.totalReceivedFees -= await this._getFee(payload, block);
+			isUpdated = true;
+		}
+
+		for (const trx of payload) {
+			if (trx.type === VoteTransaction.TYPE) {
+				for (const vote of (trx.asset as Asset).votes) {
+					const delegateVoteIndex = forgerInfo.votesReceived.findIndex(aVote =>
+						aVote.address.equals(Buffer.from(vote.delegateAddress, 'base64')),
+					);
+					if (delegateVoteIndex >= 0) {
+						forgerInfo.votesReceived[delegateVoteIndex].amount -= BigInt(vote.amount);
+						isUpdated = true;
+					}
+				}
+			}
+		}
+
+		if (isUpdated) {
+			await this._setForgerInfo(forgerAddress, { ...forgerInfo });
+		}
+	}
+
+	private async _getFee(payload: ReadonlyArray<TransactionJSON>, block: string): Promise<bigint> {
+		const { payload: payloadBuffer } = this.codec.decodeRawBlock(block);
+		const transactionFees = await this._channel.invoke<TransactionFees>('app:getTransactionsFees');
```

Comment:
This should be moved to `load` and done only once. As fee is not going to change runtime.